common/src/test/java/io/netty/util/concurrent/DefaultPromiseTest.java

```diff
@@ -204,6 +208,39 @@ public void testLateListenerIsOrderedCorrectlyFailure() throws InterruptedExcept
         testLateListenerIsOrderedCorrectly(fakeException());
     }
 
+    @Test
+    public void testSignalRace() {
+        final long wait = TimeUnit.NANOSECONDS.convert(10, TimeUnit.SECONDS);
+        EventExecutor executor = null;
+        try {
+            executor = new TestEventExecutor();
+
+            final int numberOfAttempts = 4096;
+            final Map<Thread, DefaultPromise<Void>> promises = new HashMap<Thread, DefaultPromise<Void>>();
+            for (int i = 0; i < numberOfAttempts; i++) {
+                final DefaultPromise<Void> promise = new DefaultPromise<Void>(executor);
+                final Thread thread = new Thread(new Runnable() {
+                    @Override
+                    public void run() {
+                        promise.setSuccess(null);
+                    }
+                });
+                promises.put(thread, promise);
+            }
+
+            for (final Map.Entry<Thread, DefaultPromise<Void>> promise : promises.entrySet()) {
+                promise.getKey().start();
+                final long start = System.nanoTime();
+                promise.getValue().awaitUninterruptibly(wait, TimeUnit.NANOSECONDS);
+                assertThat(System.nanoTime() - start, lessThan(wait));
+            }
+        } finally {
+            if (executor != null) {
+                executor.shutdownGracefully();
```

Comment:
Its not that it is something we never want todo but also not something super important for this sort of benchmark (as we not sync() on the future anyway).

> On 12 Jul 2016, at 18:15, Scott Mitchell <notifications@github.com> wrote:
> 
> In common/src/test/java/io/netty/util/concurrent/DefaultPromiseTest.java <https://github.com/netty/netty/pull/5523#discussion_r70470647>:
> 
> > +                    public void run() {
> > +                        promise.setSuccess(null);
> > +                    }
> > +                });
> > +                promises.put(thread, promise);
> > +            }
> > +
> > +            for (final Map.Entry<Thread, DefaultPromise<Void>> promise : promises.entrySet()) {
> > +                promise.getKey().start();
> > +                final long start = System.nanoTime();
> > +                promise.getValue().awaitUninterruptibly(wait, TimeUnit.NANOSECONDS);
> > +                assertThat(System.nanoTime() - start, lessThan(wait));
> > +            }
> > +        } finally {
> > +            if (executor != null) {
> > +                executor.shutdownGracefully();
> can you explain why we wouldn't want to do this @normanmaurer <https://github.com/normanmaurer> ?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/netty/netty/pull/5523/files/032b0e3dedadd4cdb90357b107168922713142be#r70470647>, or mute the thread <https://github.com/notifications/unsubscribe/AAa0QonOntaQcJxBKFlQ7UjeRkWcWREmks5qU72fgaJpZM4JJxnF>.
>